brainrender_napari/widgets/atlas_manager_view.py

```diff
@@ -59,20 +59,20 @@ def __init__(self, parent: QWidget = None):
 
     def _on_row_double_clicked(self):
         atlas_name = self.selected_atlas_name()
-        if atlas_name in get_downloaded_atlases():
-            up_to_date = get_atlases_lastversions()[atlas_name]["updated"]
+        if atlas_name in self.cached_downloaded_atlases:  # Use cached version
+            up_to_date = self.cached_atlas_versions[atlas_name]["updated"]
             if not up_to_date:
                 update_dialog = AtlasManagerDialog(atlas_name, "Update")
                 update_dialog.ok_button.clicked.connect(
                     self._on_update_atlas_confirmed
                 )
-                update_dialog.exec()
+                update_dialog.open()  # Use .open() instead of .exec()
```

Comment:
I am confused by this change. I don't think it should matter too much, so maybe we should not change this, in this PR at least.

Also in your PR description you say

> The .open() method has been replaced with .exec() to ensure the dialogs are modal and block execution until closed.

which is the opposite of what you do here. Can you explain?